### DISCUSSION TITLE
Request for File Encryption Feature

### DISCUSSION THREAD

**Discussion by ajeshbs** [tgdrive/teldrive#33] [Ideas]
As a feature request, I'd like to propose the addition of encryption for the files we upload via the web UI. Essentially, this would modify the file extension into a separate format when it's uploaded on the Telegram web UI. This would provide an extra layer of security, as it would make it impossible for any unauthorized individuals who gain access to our Telegram chats to open or understand our files.

Thanks

**Comment by divyam234** [comment]
@ajeshbs that possible but telegram also encrypts data  it will make uploads and downloads hella slow and consume more CPU but you can try rclone and add crypt backend on  top of teldrive but UI will be unusable after that

**Comment by DrMxrcy** [comment]
@divyam234 How can we enable encryption through the docker compose
